pkg/strategy/atrpin/strategy.go

```diff
@@ -133,25 +154,34 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
 
 		var orderForms []types.SubmitOrder
 
-		position := s.Strategy.OrderExecutor.Position()
-		s.logger.Infof("position: %+v", position)
+		baseQuantity := baseBalance.Available.Sub(s.ExpectedBaseBalance)
+		if s.TakeProfitByPosition {
+			position := s.Strategy.OrderExecutor.Position()
+			s.logger.Infof("position: %+v", position)
+			baseQuantity = position.Base
+		}
+
+		side := types.SideTypeSell
+		takerPrice := ticker.Buy
+		if baseQuantity.Sign() < 0 {
+			side = types.SideTypeBuy
+			takerPrice = ticker.Sell
+			baseQuantity = baseQuantity.Abs()
```

Comment:
I think you can check the ClosePosition() method, and maybe we can reuse some of them

Reply:
If there is a missing trade, then the position base will be incorrect. Thus I'd like to add the `ExpectedBaseBalance` to solve the problem.

Reply:
can we use query trades to fix the position?

the balance balance could be fixed by xalign, and we will need a correct PnL record

Reply:
we can't fix the balance by xalign, because atrpin will place order

Reply:
how to query trades?